### PR TITLE
 #166575772 add a reusable input component

### PR DESCRIPTION
**Pull request by dusmel**
#### What does this PR do?
This PR will add a flexible and customizable input component.

#### Description of Task to be completed?
• handle properties like required, types and placeholder
• comes with a built-in label support 
• support validation with clear messages ( The validation is triggered when the input loses focus)

#### How should this be manually tested?

| attribute  |  options | required  |
|--|--|-
| name | [value]  | true
| type | [text, email, password....]  | optional( default --> `text` )
| label | [value]  | optional
| required | [none, false]  | optional( default --> false )

#### Any background context you want to provide?

Most of the time the input is handled multiple times with a different design. 
Since inputs are going to be used extensively in the app it is logic and powerful to have a reusable component that is flexible and customizable hence the implementation of this chore.

#### What are the relevant pivotal tracker stories?
[#166575772](https://www.pivotaltracker.com/story/show/166575772)

#### Screenshots (if appropriate)
# 1
image

# 2
image

# 3
image

# 4
image

#### Questions:

N/A

### CODE REVIEW

**Review by coolbeatz71 — CHANGES_REQUESTED** [review]
Excellent work! @dusmel. Just one thing to take a look. In my opinion, putting the `orange` background on the `error message` doesn't look good. Once we get a lot of error messages on the form, It wouldn't look nice as the orange color will be dominant, thus probably bad for the user.

For example, If you could just put a red text, It will be sufficient and nice for the user.

Apart from the error message, everything else looks 👍👍👍

**Comment by Karlmusingo** [comment]
Hey @dusmel, Your PT id should be a link to the Pivotal Tracker story.

**Review by Karlmusingo — APPROVED** [review]

File: src/Components/Common/input/exemple.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      <React.fragment>
+        <Input />
+        <Input />
+        <button className="btn btn-primary">submit</button>
+      </React.fragment>
+    );
+  }
+}
+
+export default Form;
```
Comment (gracelungu) [inline src/Components/Common/input/exemple.jsx]: This example file is irrelevant, it is only needed in the development and should be removed after.

File: src/Components/Common/input/input.jsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
+import React, { Component } from 'react';
+import Joi from 'joi-browser';
+// import { connect } from 'react-redux';
```
Comment (gracelungu) [inline src/Components/Common/input/input.jsx]: Commented codes should not be present. These serve no purpose in the implementation of the project.

**Comment by dusmel** [comment]
> Excellent work! @dusmel. Just one thing to take a look. In my opinion, putting the `orange` background on the `error message` doesn't look good. Once we get a lot of error messages on the form, It wouldn't look nice as the orange color will be dominant, thus ugly.
> 
> For example, just a red text will be sufficient.
> 
> Apart from the error message, everything else looks 👍👍👍

@coolbeatz71 I get your point but the way the validation is designed it gives you the error as soon as the input loses the focus, so you will probably not have all of them at the same time

**Review by gracelungu — CHANGES_REQUESTED** [review]

**Review by gracelungu — APPROVED** [review]

**Comment by coolbeatz71** [comment]
> > Excellent work! @dusmel. Just one thing to take a look. In my opinion, putting the `orange` background on the `error message` doesn't look good. Once we get a lot of error messages on the form, It wouldn't look nice as the orange color will be dominant, thus ugly.
> > For example, just a red text will be sufficient.
> > Apart from the error message, everything else looks 👍👍👍
> 
> @coolbeatz71 I get your point but the way the validation is designed it gives you the error as soon as the input loses the focus, so you will probably not have all of them at the same time

Okay

**Comment by coolbeatz71** [comment]
> Excellent work! @dusmel. Just one thing to take a look. In my opinion, putting the `orange` background on the `error message` doesn't look good. Once we get a lot of error messages on the form, It wouldn't look nice as the orange color will be dominant, thus ugly.
> 
> For example, just a red text will be sufficient.
> 
> Apart from the error message, everything else looks 👍👍👍

Ok

**Review by coolbeatz71 — APPROVED** [review]

**Comment by dkam26** [comment]
@dusmel , well done! How come you didn't write tests for the component? Kindly write tests for the component!

**Pull request merged**